src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -406,24 +408,58 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
          1. RequestPriority
          2. Global workqueue statistics, while acquiring a workflow
         """
-        if 'RequestPriority' in request_args:
-            # Yes, we completely ignore any other arguments posted by the user (web UI case)
-            request_args = {'RequestPriority': request_args['RequestPriority']}
-            validate_request_priority(request_args)
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
+        report = []
+        reqArgsDiff = deepcopy(request_args)
+
+        # NOTE: Here we need to re-add `RequestName` since down bellow will be the second call of `validate_request_update_args`
+        #       which, by itself is a destructive function, which changes `request_args` with the following line:
+        #           request_name = request_args.pop("RequestName")
+        #       So if we call this function more than once inside the same request instance on every next call it will raise:
+        #           `KeyError: 'RequestName'`.
+        #       The first time it was called was during the validation of the REST call.
+        if 'RequestName' not in reqArgsDiff:
+            reqArgsDiff['RequestName'] = workload.name()
+        _, reqArgsDiff = validate_request_update_args(reqArgsDiff, self.config, self.reqmgr_db_service, None)
+
+        def _reduceReport(report):
+            if reduce(lambda x, y: x == y == 'OK' and 'OK', report, 'OK'):
+                return 'OK'
+            return False
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return _reduceReport(report)
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgsDiff)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
+                report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
```

Comment:
I was considering updating it once per every parameter instead..... with a call of the form:
```
report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), {reqArg: reqArgsDiff[reqArg]}, dn))
```
Such that we go progressively with those workflow  changes and update as much as we can at couch before we give up.... but anyway we can do it in a single go as well. I'll change that.

Reply:
about this: 
> Having a single update call keeps it simple as well and does not require the reduce custom function that you implemented above

We would actually still need it, because we are aggregating reports from two different calls to couchdb:
 * `reqmgr_db_service.updateRequestStats`
 * `reqmgr_db_service.updateRequestProperty`